IdentityCore/src/webview/operations/MSIDWebResponseBrokerInstallOperation.m

```diff
@@ -0,0 +1,100 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#import "MSIDWebResponseBrokerInstallOperation.h"
+#import "MSIDWebviewResponse.h"
+#import "MSIDBrokerInteractiveController.h"
+#import "MSIDInteractiveTokenRequestParameters.h"
+#import "MSIDWebWPJResponse.h"
+#import "MSIDTokenRequestProviding.h"
+#import "MSIDKeychainTokenCache.h"
+#import "MSIDLegacyTokenCacheAccessor.h"
+#import "MSIDDefaultTokenCacheAccessor.h"
+#import "MSIDAADV2Oauth2Factory.h"
+#import "MSIDDefaultTokenResponseValidator.h"
+#import "MSIDAccountMetadataCacheAccessor.h"
+
+@interface MSIDWebResponseBrokerInstallOperation()
+
+@property (nonatomic) NSURL *appInstallLink;
+
+@end
+
+@implementation MSIDWebResponseBrokerInstallOperation
+
+- (nullable instancetype)initWithResponse:(MSIDWebviewResponse *)response
+                                    error:(NSError **)error
+{
+    #if TARGET_OS_IPHONE
+        self = [super initWithResponse:response
+                                 error:error];
+        if (self)
+        {
+            if (![response isKindOfClass:MSIDWebWPJResponse.class] || [NSString msidIsStringNilOrBlank:[(MSIDWebWPJResponse *)response appInstallLink]])
+            {
+                return nil;
+            }
+            
+            MSIDWebWPJResponse *wpjResponse = (MSIDWebWPJResponse *)response;
+            _appInstallLink = [NSURL URLWithString:wpjResponse.appInstallLink];
+        }
+        
+        return self;
+    #else
+        return nil;
+    #endif
+}
+
+- (void)invokeWithInteractiveTokenRequestParameters:(MSIDInteractiveRequestParameters *)interactiveTokenRequestParameters
+                               tokenRequestProvider:(id<MSIDTokenRequestProviding>)tokenRequestProvider
+                                         completion:(MSIDRequestCompletionBlock)completion
+{
+    if (!completion)
+    {
+        return;
+    }
+    
+    #if TARGET_OS_IPHONE
+        if ([interactiveTokenRequestParameters isKindOfClass:MSIDInteractiveTokenRequestParameters.class])
+        {
+            NSError *brokerError;
+            MSIDBrokerInteractiveController *brokerController = [[MSIDBrokerInteractiveController alloc] initWithInteractiveRequestParameters:(MSIDInteractiveTokenRequestParameters *)interactiveTokenRequestParameters
```

Comment:
check broker error here and return early if you received one

Reply:
updated